pyhive/hive.py

```diff
@@ -79,20 +82,29 @@ def __init__(self, host, port=10000, username=None, database='default', auth='NO
         if auth == 'NOSASL':
             # NOSASL corresponds to hive.server2.authentication=NOSASL in hive-site.xml
             self._transport = thrift.transport.TTransport.TBufferedTransport(socket)
-        elif auth == 'NONE':
+        elif auth.upper() in ['LDAP', 'PLAIN', 'GSSAPI', 'NONE']:
+            if auth.upper() == 'NONE':
+                #Follow the older version's convention
+                auth = 'PLAIN'
+            if password is None:
+                if auth == 'LDAP':
+                    password = ''
+                else:
+                    # PLAIN always requires a password for HS2.
+                    password = 'x'
             def sasl_factory():
                 sasl_client = sasl.Client()
-                sasl_client.setAttr(b'username', username.encode('latin-1'))
-                # Password doesn't matter in NONE mode, just needs to be nonempty.
-                sasl_client.setAttr(b'password', b'x')
+                sasl_client.setAttr('host', host)
+                sasl_client.setAttr('service', kerberos_service_name)
+                if auth.upper() in ['PLAIN', 'LDAP']:
+                    sasl_client.setAttr('username', user)
```

Comment:
I don't see `user` defined here. Also what happens if `user` is passed as a unicode string?